src/storage-client/src/storage_collections.rs

```diff
@@ -2243,7 +2242,9 @@ where
 
         let mut txns_upper_future = match self.txns_handle.take() {
             Some(txns_handle) => {
-                let txns_upper_future = gen_upper_future(GlobalId::Transient(1), txns_handle);
+                let upper = txns_handle.upper().clone();
```

Comment:
Do we think `upper()` is up-to-date enough compared to `shared_upper()`, which we used before?

Reply:
No idea! Do you know what "up-to-date enough" means? I was under the assumption that we could in principle also use `Antichain::from_elem(T::minimum())` here and it would be fine, since `update_write_frontier` handles updates that are less than the current `collection.write_frontier` gracefully (by ignoring them). Did I miss something?

Using `upper()` here seems less risky than `shared_upper()` because the latter opens us up to hangs when the shard advances to the empty frontier concurrently. That probably can't happen for collections in the txns system currently, but still... things can change.

Reply:
Yeah, that's a good idea, maybe using `upper()` here then is more conservative, and safer!